### PR TITLE
EVG-19680 Add logging for PR mergable state 

### PR DESCRIPTION
**Pull request by malikchaya2**
EVG-19680

### Description
Logging the state of PRs we try to merge will help us with debugging.

### CODE REVIEW

**Review by hadjri — COMMENTED** [review]
Do we have a ticket to undo this once investigation is done?

File: rest/data/commit_queue.go [hunk 1]
```diff
@@ -373,10 +373,19 @@ func getPRAndCheckMergeable(ctx context.Context, env evergreen.Environment, sc C
 	if err != nil {
 		return nil, err
 	}
-
+	mergeableState := pr.GetMergeableState()
+
+	grip.Info(message.WrapError(err, message.Fields{
```
Comment (ablack12) [inline rest/data/commit_queue.go:378]: If you wrap error here, you'll only actually log if the error isn't nil -- not sure if that's what you want.

**Review by ablack12 — APPROVED** [review]
LGTM with note

File: rest/data/commit_queue.go (hunk 1 above)
Comment (ablack12) [reply rest/data/commit_queue.go:378]: Can we also use grip.Debug since this is for debugging?

**Review by hadjri — APPROVED** [review]

**Pull request merged**